Tasks/Maven/pmdformaven.ts

```diff
@@ -37,43 +39,52 @@ export function processPmdOutput(sourcesDirectory:string) : ar.AnalysisResult {
 
 // Verifies the existence of the HTML output file.
 // Modifies the relevant field within the returned object accordingly.
-// Task fails if the file cannot be found.
 function processPmdHtml(analysisResult:ar.AnalysisResult, path:string):ar.AnalysisResult {
-    // Task fails if the file cannot be found.
     if (!tl.exist(path)) {
-        throw new Error("Could not find PMD HTML output. Check that PMD ran successfully and that it is writing to the default location.");
+        tl.debug('PMD HTML not found at ' + path);
+    } else {
+        analysisResult.htmlFilePath = path;
     }
-    analysisResult.htmlFilePath = path;
     return analysisResult;
 }
 
 // Verifies the existence of the XML output file and parses its contents.
 // Modifies the relevant fields within the returned object accordingly.
-// Task fails if the file cannot be found.
 function processPmdXml(analysisResult:ar.AnalysisResult, path:string):ar.AnalysisResult {
     if (!tl.exist(path)) {
-        throw new Error("Could not find PMD XML output. Check that PMD ran successfully and that it is writing to the default location.");
+        tl.debug('PMD XML not found at ' + path);
     }
 
     var pmdXmlFileContents = fs.readFileSync(path, 'utf-8');
     xml2js.parseString(pmdXmlFileContents, function (err, data) {
-        if (data) {
-            analysisResult.xmlFilePath = path;
+        if (!data) { // Not an XML file, throw error
+            throw new Error("Failed to parse XML output from PMD.");
```

Comment:
As discussed, the handling should be consistent with the error case in lines 63-65 below - either ignore or throw in both cases. Should probably throw - we're not expecting it to happen, and if we don't throw we won't know that something is wrong.